elasticsearch_dsl/faceted_search.py

```diff
@@ -47,13 +56,28 @@ def facets(self):
             for name, agg in iteritems(self._search.facets):
                 buckets = self._facets[name] = []
                 data = self.aggregations['_filter_' + name][name]['buckets']
-                filter = self._search._raw_filters.get(name, ())
+                filter = self._search._raw_filters.get(name, {})
                 for b in data:
-                    buckets.append(BUCKET_TO_DATA[agg.__class__](b, filter))
+                    agg_class = agg.__class__
+                    if agg_class == Range:
```

Comment:
Classes should be compared by identity. I don't think this `if` branch belongs here, it's actually converting range-specific buckets to data so it deserves to live in `BUCKET_TO_DATA`.